### ISSUE TITLE
"loading..." message can cause confusion

### ISSUE THREAD

**Issue by shiffman** [processing/p5.js#764]
I don't recall if there's already an issue or discussion on this but #763 reminded me of something that came up several times in a recent p5.js workshop.   The "loading..." message is intended for the case when  `preload()` is doing its work for large files.   Instead of thinking there might be an error, a user knows "ok, my sketch is just loading."  However, whenever there is a JS error before things get going, a p5 sketch will sit on "loading..." This can have the opposite effect and discourage users from thinking there might be a syntax error or problem with their code.

I don't know how to resolve this and I don't love any of the ideas I can think of like (a) doing away with it or (b) having it animate when it really is loading? or (c) if enough time has passed p5 could show something else like "This sketch hasn't started up and it's been quite a well, maybe check the console?"

Does anyone else encounter this or have any thoughts?

**Comment by joecridge** [comment]
This has vexed me a couple times when I have been working on p5 sketches without devtools open. I like the idea of animating e.g. the ellipsis, since developers will notice that something is hanging and it might prompt them to investigate, yet it wouldn't be to distracting for end users when the message is displayed as intended.

**Comment by lmccart** [comment]
so maybe the solution is to only show the load message if preload is defined, rather than always by default. this would still hang on errors but only errors within preload.

**Comment by futuremarc** [comment]
This issue is actually always on my mind. I like @lmccart 's solution, except can we find a graceful way to display an error when working with preload? I feel like when I'm working on a sketch with preload, there's always this anticipation if 'loading...' means its broken or if the sketch is actually loading.

**Comment by lmccart** [comment]
what would you suggest? I worry that displaying errors directly on the page may be going too far vs just directing developers to the console or the IDEs?

**Comment by futuremarc** [comment]
How were we thinking of displaying the friendly error messages, keeping them in the console?

**Comment by lmccart** [comment]
yes. the IDEs might interpret and do something more with them.

**Comment by lmccart** [comment]
Ok I changed things around so the "Loading..." div is only added if preload is defined, and it is removed immediately after preload completes. If there is no preload defined, we will never see the loading div at all. So it could theoretically hang if you mistype the path of a file or a function within preload, but I think this is a big improvement over previously when it could hang on "Loading..." anytime there was any typo anywhere in the sketch.

**Comment by joecridge** [comment]
Great :)

**Comment by futuremarc** [comment]
Woo!

**Comment by dhowe** [comment]
How about updating the 'loading' div with an error when if/when we get one?

**Comment by lmccart** [comment]
love it

**Comment by jywarren** [comment]
Hi, I found something odd, that running this actually produces an element called "<loadingdiv>":

![image](https://user-images.githubusercontent.com/24359/85970986-03518e00-b99a-11ea-8fca-35939f313a85.png)

It came up because I was trying to create CSS to hide the loadingdiv which in my app I don't want to see. So, for posterity, if folks want to, the CSS required ought to be:

```css
loadingdiv { display:none; }
```

However, maybe because of order of execution, this worked for ones I generated from the console with `lmccart`'s code, but not for those displaying on pageload. I even tried putting the CSS in the document header but somehow it's not interpreted fast enough, perhaps?

Anyways no big deal. Thanks!
